api/v1beta2/network_types.go

```diff
@@ -307,8 +308,18 @@ func (v *VPCSpec) IsIPv6Enabled() bool {
 // SubnetSpec configures an AWS Subnet.
 type SubnetSpec struct {
 	// ID defines a unique identifier to reference this resource.
+	// If you're bringing your subnet, set the AWS subnet-id here, it must start with `subnet-`.
+	//
+	// If you'd like the provider to create a subnet for you, the id can be set to
+	// any placeholder value that does not start with `subnet-`; upon creation, the provider will
+	// replace the placeholder with the actual subnet id and set the ARN field.
 	ID string `json:"id"`
 
+	// ResourceID is the subnet identifier from AWS, READ ONLY.
+	// This field is populated when the provider manages the subnet.
```

Comment:
How is the read only nature of this field enforced?

Reply:
Should we add CEL then to make it immutable?
```suggestion
	// ResourceID is the subnet identifier from AWS, READ ONLY.
	// This field is populated when the provider manages the subnet.
	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="resourceID is immutable"
```

Reply:
Do we want to handle this on this PR or as a follow up?

Reply:
I don't have a strong opinion to enforce this today, the reconciler will overwrite it anyways